### ISSUE TITLE
make-ca-cert.sh: line 31: 1: unbound variable

### ISSUE THREAD

**Issue by choclo** [kubernetes/kubernetes#11743]
Hi guys,

I'm trying to install kubernetes on a Fedora 21 (and 22) virtual machine however, every time I try the install it fails (I'm using ansible for installation of both master and nodes).
Logging of the error is as follows:

```
TASK: [kubernetes | Run create cert script on master] ************************* 
failed: [kubertest.tov.cl] => {"changed": true, "cmd": ["/usr/libexec/kubernetes/make-ca-cert.sh", "kubertest.tov.cl"], "delta": "0:00:00.393056", "end": "2015-07-23 10:22:15.324534", "rc": 2, "start": "2015-07-23 10:22:14.931478", "warnings": []}
stdout: === Failed to generate certificates: Aborting ===

FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/home/tomas/cluster.retry

kubermin1.tov.cl           : ok=21   changed=1    unreachable=0    failed=0   
kubertest.tov.cl           : ok=58   changed=1    unreachable=0    failed=1   
```

When trying to run manually I get the following:

```
[root@kubertest ~]# /usr/libexec/kubernetes/make-ca-cert.sh
/usr/libexec/kubernetes/make-ca-cert.sh: line 31: 1: unbound variable
[root@kubertest ~]# cat /etc/redhat-release 
Fedora release 21 (Twenty One)
```

Looking at the file (line 31) it appears maybe environment variables are not being set correctly (cert_ip) (well, now that I think about, I'm running it from a prompt thus, env variables may not be correct):

```
# Caller should set in the ev:
# MASTER_IP - this may be an ip or things like "_use_gce_external_ip_"
# DNS_DOMAIN - which will be passed to minions in --cluster_domain
# SERVICE_CLUSTER_IP_RANGE - where all service IPs are allocated
# MASTER_NAME - I'm not sure what it is...

# Also the following will be respected
# CERT_DIR - where to place the finished certs
# CERT_GROUP - who the group owner of the cert files should be

cert_ip="${MASTER_IP:="${1}"}"
master_name="${MASTER_NAME:="kubernetes"}"
service_range="${SERVICE_CLUSTER_IP_RANGE:="10.0.0.0/16"}"
dns_domain="${DNS_DOMAIN:="cluster.local"}"
cert_dir="${CERT_DIR:-"/srv/kubernetes"}"
cert_group="${CERT_GROUP:="kube-cert"}"
```
Just in case, group "kube-cert" DOES exist.
I tried both with and without selinux enforcing and still nothing.
Let me know if you need more info from me.

Thanks!

**Comment by jeffbean** [comment]
@eparis This looks like to me the line is not escaped correctly. 

possible fix: https://github.com/jeffbean/kubernetes/commit/ca8fb7cb4ec9c8f8c536abb65898fa2aea4ec0e7 

@choclo can you try out this patch in your case and see this is the fix.

**Comment by jeffbean** [comment]
@choclo Also to know what ENV vars are being passed to this script via Ansible. You can run the playbook with higher verbosity. This will ensure that what you are expecting is correct during the run of the playbook.

`./setup.sh -vvv`

**Comment by jeffbean** [comment]
I changed the edit to prefer the argument over the ENV var. read up on the default substitution. In this case we could do no argument but this works for now.

**Comment by choclo** [comment]
Unfortunately this is still not working @jeffbean even with fix @ca8fb7c

```
TASK: [kubernetes | Run create cert script on master] ************************* 
<kubertest.tov.cl> ESTABLISH CONNECTION FOR USER: root
<kubertest.tov.cl> REMOTE_MODULE command creates=/etc/kubernetes/certs/server.crt /usr/libexec/kubernetes/make-ca-cert.sh kubertest.tov.cl
<kubertest.tov.cl> EXEC ssh -C -tt -v -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/home/tomas/.ansible/cp/ansible-ssh-%h-%p-%r" -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 kubertest.tov.cl /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1437721240.5-105634379282919 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1437721240.5-105634379282919 && echo $HOME/.ansible/tmp/ansible-tmp-1437721240.5-105634379282919'
<kubertest.tov.cl> PUT /tmp/tmpr3HMtI TO /root/.ansible/tmp/ansible-tmp-1437721240.5-105634379282919/command
<kubertest.tov.cl> EXEC ssh -C -tt -v -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/home/tomas/.ansible/cp/ansible-ssh-%h-%p-%r" -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 kubertest.tov.cl /bin/sh -c 'sudo -k && sudo -H -S -p "[sudo via ansible, key=uwpdxubmychdjccemlnropvucsjxzzsc] password: " -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-uwpdxubmychdjccemlnropvucsjxzzsc; LANG=C CERT_DIR=/etc/kubernetes/certs MASTER_NAME=kubernetes LC_CTYPE=C DNS_DOMAIN=cluster.local MASTER_IP=46.101.35.191 CERT_GROUP=kube-cert SERVICE_CLUSTER_IP_RANGE=10.254.0.0/16 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1437721240.5-105634379282919/command; rm -rf /root/.ansible/tmp/ansible-tmp-1437721240.5-105634379282919/ >/dev/null 2>&1'"'"''
failed: [kubertest.tov.cl] => {"changed": true, "cmd": ["/usr/libexec/kubernetes/make-ca-cert.sh", "kubertest.tov.cl"], "delta": "0:00:00.184416", "end": "2015-07-24 07:00:40.713656", "rc": 2, "start": "2015-07-24 07:00:40.529240", "warnings": []}
stdout: === Failed to generate certificates: Aborting ===

FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/home/tomas/cluster.retry

kubermin1.tov.cl           : ok=21   changed=1    unreachable=0    failed=0   
kubertest.tov.cl           : ok=58   changed=1    unreachable=0    failed=1   
```

**Comment by eparis** [comment]
So @jeffbean definitely is fixing a bug, but apparently not your bug. My guess? Do you have openssl installed on the master?
https://github.com/eparis/kubernetes/commit/fd1024baa2c70c248cc5d051c8a400bd6fffabe8

If that doesn't fix it you can remove all of the `> /dev/null 2>&1` portions at the end of all of the `easyrsa` lines. Should give us more debug info (assuming its just not openssl not being installed)

**Comment by choclo** [comment]
Actually, it was openssl not being installed (NOW I feel stupid).
So eparis@fd1024b should be super helpful.

Thanks @eparis!

**Comment by jeffbean** [comment]
This is already fixed in our PR over here: #11264

**Comment by roberthbailey** [comment]
Looks like this was fixed by #11264. Please reopen if you are still having this issue.

**Comment by tech2free** [comment]
Actually This is failed because no server.key and server.crt was generated from the downloaded tar.gz : 
https://storage.googleapis.com/kubernetes-release/easy-rsa/easy-rsa.tar.gz
